server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -297,6 +300,23 @@ private static void sendErrorResponse(String actionName, TransportChannel transp
         try {
             response = handler.read(stream);
             response.remoteAddress(new TransportAddress(remoteAddress));
+
+            if (stream != EMPTY_STREAM_INPUT) {
+                // Check the entire message has been read
+                final int nextByte = stream.read();
```

Comment:
I realize the exception messages are slightly different, but is it possible to pull this logic out into some kind of `assertFullyReady()` helper method since it is done in 3 different places?

Reply:
:+1: